lib/elixir/lib/task/supervised.ex

```diff
@@ -10,26 +12,37 @@ defmodule Task.Supervised do
   end
 
   def start_link(caller, link, info, fun) do
-    :proc_lib.start_link(__MODULE__, :reply, [caller, link, info, fun])
+    {:ok, spawn_link(caller, link, info, fun)}
   end
 
-  def async(caller, info, mfa) do
-    initial_call(mfa)
-    ref = receive do: ({^caller, ref} -> ref)
-    send caller, {ref, do_apply(info, mfa)}
+  def spawn_link(caller, link \\ :nolink, info, fun) do
+    :proc_lib.spawn_link(__MODULE__, :reply, [caller, link, info, fun])
```

Comment:
Also even if we replicated the initial call and ancestors I am unsure if all tools will use them if the actual initial call is not `:proc_lib`.